metrics-core/src/traits.rs

```diff
@@ -51,9 +63,16 @@ pub trait Gauge: MetricCommon {
 
 /// Any sort of summary of the record values
 pub trait Summary: MetricCommon {
+    // TODO: function parameters for record are inconsistent with other traits
+
     /// Record `count` instances of `val`.
     fn record(&self, time: Instant, val: u64, count: u64);
 
+    // TODO: we should consider alternate design here. It's more likely that
+    // we'd want to get some preset group of submetrics (eg: particular
+    // quantiles) it's unclear how we'd tell the implementation about those in
+    // the current form.
+
     /// Get all statistics exposed by the implementation
     fn submetrics(&self) -> Vec<SubMetric>;
```

Comment:
I think that we could do with a redesign of the `Summary` trait. I agree that the interface provided here isn't very good.